pset/pset.go

```diff
@@ -283,6 +289,200 @@ func (p *Pset) SanityCheck() error {
 	return nil
 }
 
+func (p *Pset) ValidateAllSignatures() (bool, error) {
+	for i := range p.Inputs {
+		valid, err := p.ValidateInputSignatures(i)
+		if err != nil {
+			return false, err
+		}
+		if !valid {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (p *Pset) ValidateInputSignatures(inputIndex int) (bool, error) {
+	for _, partialSig := range p.Inputs[inputIndex].PartialSigs {
```

Comment:
it should be better to first check that the input actually contains some signature, otherwise this method would return true for inputs that don't have any

Reply:
Done